apis/v1/gateway_types.go

```diff
@@ -300,10 +300,27 @@ type Listener struct {
 	//
 	// * TLS: The Listener Hostname MUST match the SNI.
 	// * HTTP: The Listener Hostname MUST match the Host header of the request.
-	// * HTTPS: The Listener Hostname SHOULD match at both the TLS and HTTP
-	//   protocol layers as described above. If an implementation does not
-	//   ensure that both the SNI and Host header match the Listener hostname,
-	//   it MUST clearly document that.
+	// * HTTPS: The Listener Hostname SHOULD match both the SNI and Host header.
+	//   Note that this does not require the SNI and Host header to be the same.
+	//   The semantics of this are described in more detail below.
+	//
+	// To ensure security, Section 11.1 of RFC-6066 emphasizes that server
+	// implementations that rely on SNI hostnames matching must also verify
+	// hostnames within the application protocol.
+	//
+	// Section 9.1.2 of RFC-7540 provides a mechanism for servers to reject the
+	// reuse of a connection by responding with the HTTP 421 Misdirected Request
+	// status code. This indicates that the origin server has rejected the
+	// request because it appears to have been misdirected.
+	//
+	// To detect misdirected requests, Gateways SHOULD match the authority of
+	// the requests with all the SNI hostname(s) configured across all the
+	// Gateway Listeners on the same port:
+	//
+	// * If another Listener has an exact match or more specific wildcard entry,
+	//   the Gateway should return a 421.
+	// * If the current Listener doesn’t match the SNI or Host, the reverse
+	//   proxy should return a 421.
```

Comment:
```suggestion
	// * If the current Listener doesn’t match the SNI or Host, the Gateway
	// should return a 421.
```

Reply:
(Making @howardjohn's comment above easier to accept 😉)

Reply:
Updated, thanks!